src/Sentry.Unity/ScriptableSentryUnityOptions.cs

```diff
@@ -0,0 +1,94 @@
+using Sentry.Unity.Extensions;
+using UnityEngine;
+
+namespace Sentry.Unity
+{
+    public class ScriptableSentryUnityOptions : ScriptableObject
+    {
+        /// <summary>
+        /// Relative to Assets/Resources
+        /// </summary>
+        public const string ConfigRootFolder = "Sentry";
+
+        /// <summary>
+        /// Main Sentry config name for Unity
+        /// </summary>
+        public const string ConfigName = "SentryOptions";
+
+        /// <summary>
+        /// Path for the config for Unity
+        /// </summary>
+        internal static string GetConfigPath(string? notDefaultConfigName = null)
+            => $"Assets/Resources/{ConfigRootFolder}/{notDefaultConfigName ?? ConfigName}.asset";
+
+        [field: SerializeField] public bool Enabled { get; set; }
+
+        [field: SerializeField] public bool CaptureInEditor { get; set; }
+
+        [field: SerializeField] public string Dsn { get; set; } = "";
+
+        [field: SerializeField] public float SampleRate { get; set; } = 1.0f;
+
+        [field: SerializeField] public bool AttachStacktrace { get; set; }
+        [field: SerializeField] public string ReleaseOverride { get; set; } = "";
+        [field: SerializeField] public string EnvironmentOverride { get; set; } = "";
+        [field: SerializeField] public bool EnableOfflineCaching { get; set; }
+
+        [field: SerializeField] public bool Debug { get; set; }
+        [field: SerializeField] public bool DebugOnlyInEditor { get; set; }
+        [field: SerializeField] public SentryLevel DiagnosticLevel { get; set; }
+
+        public static SentryUnityOptions? LoadSentryUnityOptions()
+        {
+            // TODO: Deprecated and to be removed once we update far enough.
+            var sentryOptionsTextAsset = Resources.Load<TextAsset>($"{ConfigRootFolder}/{ConfigName}");
+            if (sentryOptionsTextAsset != null)
+            {
+                var options = JsonSentryUnityOptions.LoadFromJson(sentryOptionsTextAsset);
+                return options;
+            }
+
+            var scriptableOptions = Resources.Load<ScriptableSentryUnityOptions>($"{ConfigRootFolder}/{ConfigName}");
+            if (scriptableOptions != null)
+            {
+                return LoadFromSerializableObject(scriptableOptions);
+            }
+
+            return null;
+        }
+
+        private static SentryUnityOptions? LoadFromSerializableObject(ScriptableSentryUnityOptions scriptableOptions)
+        {
+            var options = new SentryUnityOptions();
+            SentryOptionsUtility.SetDefaults(options);
+
+            options.Enabled = scriptableOptions.Enabled;
+            options.CaptureInEditor = scriptableOptions.CaptureInEditor;
+            options.Dsn = scriptableOptions.Dsn;
+            options.SampleRate = scriptableOptions.SampleRate;
+            options.AttachStacktrace = scriptableOptions.AttachStacktrace;
+
+            if (!string.IsNullOrEmpty(scriptableOptions.ReleaseOverride))
```

Comment:
We should probably replace all `string.IsNullOrEmpty` with `string.IsNullOrWhiteSpace` calls.